code/controllers/subsystem/library.dm

```diff
@@ -49,7 +49,8 @@ SUBSYSTEM_DEF(library)
 /datum/controller/subsystem/library/proc/prepare_official_posters()
 	printable_posters = list()
 	for(var/obj/structure/sign/poster/official/poster_type as anything in subtypesof(/obj/structure/sign/poster/official))
-		printable_posters[initial(poster_type.name)] = poster_type
+		if (!printable_posters[initial(poster_type.name)]) //prevent overwriting the plain version with the directional version of posters (mainly for the /random one)
+			printable_posters[initial(poster_type.name)] = poster_type
```

Comment:
depending on the output order of typesof is concerning

Reply:
It is concerning, do you have suggestions to properly get *only* subtypes and not sub-subtypes?

Reply:
Check if `initial(poster_type.parent_type) == /obj/structure/sign/poster/official`.

We definitely should not be doing this as typesof behavior is not only unspecified, but also has changed in the past, causing https://github.com/tgstation/tgstation/pull/65241

Reply:
oh that's a much better solution then mine

Reply:
Will add this soon, thanks for the suggestion!

Reply:
Ok this should be fixed now. Initial(type.parent_type) doesn't work because reasons so I had to go with an alternative suggested by Lemon.